File: software/examples/javascript/ExampleSimple.js
var IPConnection = require('Tinkerforge/IPConnection');
var BrickletMultiTouch = require('Tinkerforge/BrickletMultiTouch');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'jQs';// Change to your UID

var ipcon = new IPConnection();// Create IP connection
var mt = new BrickletMultiTouch(UID, ipcon);// Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);        
    }
);// Connect to brickd

// Don't use device before ipcon is connected
ipcon.on(IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        //Get current touch state
        mt.getTouchState(
            function(touchState) {
                var s = '';
                if (touchState & (1 << 12)) {
                    s += 'In proximity, ';
                }
                if ((touchState & 0xFFF) === 0) {
                    s += 'No electrodes touched\n'
                }
                else {
                    s += 'Electrodes ';
                    for(var i=0; i<12; i++) {
                        if(touchState & (1 << i)) {
                            s += parseInt(i)+' ';
                        }
                    }
                    s += 'touched\n'
                }
                console.log(s);
            },
            function(error) {
                console.log('Error: '+error);
            }
        );
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{ipconnection::IpConnection, multi_touch_bricklet::*};

const HOST: &str = "127.0.0.1";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Multi Touch Bricklet

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection
    let multi_touch_bricklet = MultiTouchBricklet::new(UID, &ipcon); // Create device object

    ipcon.connect(HOST, PORT).recv()??; // Connect to brickd
                                        // Don't use device before ipcon is connected

    // Get current touch state
    let state = multi_touch_bricklet.get_touch_state().recv()?;
    let mut string = String::new();

    if (state & (1 << 12)) == (1 << 12) {
        string.push_str("In proximity, ");
    }

    if (state & 0xfff) == 0 {
        string.push_str("No electrodes touched");
    } else {
        string.push_str("Electrodes ");
        for i in 0..12 {
            if (state & (1 << i)) == (1 << i) {
                string.push_str(&(i.to_string() + " "));
            }
        }
        string.push_str("touched");
    }

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
